File: Data Structures & Algorithms/validate-parentheses/submission-3.cpp
class Solution {
public:
    bool isValid(string s) {
        stack <char> charStrack;


        for(int i=0;s[i];i++){
            if(s[i] == '(' || s[i] == '{' || s[i] == '['){
                charStrack.push(s[i]);
            }
            else if(charStrack.empty()){
                return false;
            }

            else if((s[i] == ')' && charStrack.top() == '(') || (s[i] == ']' && charStrack.top() == '[') || (s[i] == '}' && charStrack.top() == '{')){
                charStrack.pop();
            }
            else{
                return false;
            }
            
        }
        return charStrack.empty();
    }
};

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-3.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        answer = 0
        for char in tokens:
            if char in {"+", "-", "*", "/"}:
                second = stack.pop()
                first = stack.pop()
                if char == '+':
                    stack.append(first + second)
                elif char == '-':
                    stack.append(first - second)
                elif char == '*':
                    stack.append(first * second)
                elif char == '/':
                    stack.append(int(first / second))
            else:
                stack.append(int(char))
        return stack.pop()